content/developer/howtos/upgrade/upgrade_util.rst

```diff
@@ -0,0 +1,283 @@
+.. method:: merge_model(cr, source, target[, drop_table=True][, fields_mapping=None][, ignore_m2m=()])
+
+   Merge the references from ``source`` model into ``target`` model and removes ``source`` model and
+   its references. By default, only the fields with the same name in both models are mapped.
+
+   .. warning::
+      This function does not move the records from ``source`` model to ``target`` model.
+
+   :param str source: name of the source model of the merge
+   :param str target: name of the destination model of the merge
+   :param bool drop_table: if ``True``, drops the source model's table (default: ``True``)
+   :param dict fields_mapping: Dictionary ``{"source_model_field_1": "target_model_field_1", ...}``
+      mapping fields with different names on both models
+   :param list(str) or str ignore_m2m: list of m2m tables ignored to remove from source model.
+
+Modules
+-------
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/modules.py#L218>`_
+.. method:: remove_module(cr, module)
+
+   Uninstall and remove a module and its references from the database
+
+   :param str module: name of the module to remove
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/modules.py#L263>`_
+.. method:: rename_module(cr, old, new)
+
+   Rename a module and its references from ``old`` to ``new``
+
+   :param str old: current name of the module to rename
+   :param str new: new name of the module to rename
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/modules.py#L323>`_
+.. method:: merge_module(cr, old, into, update_dependers=True)
+
+   Move all references of module ``old`` into module ``into``
+
+   :param str old: name of the source module of the merge
+   :param str into: ame of the destination module of the merge
+   :param bool update_dependers: if ``True``, updates the dependencies of modules that depends on
+      ``old`` (default: ``True``)
+
+ORM
+---
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/orm.py#L43>`_
+.. method:: env(cr)
+
+   Create a new environment from the cursor
+
+   .. warning::
+      This function does NOT empty the cache maintained on the cursor for superuser with an empty
+      environment. A call to invalidate_cache will most probably be necessary every time you
+      directly modify something in database.
+
+   :return: The new environment
+   :rtype: :class:`~odoo.api.Environment`
+
+.. `[source] <https://github.com/odoo/upgrade-util/blob/master/src/util/orm.py#L218>`_
+.. method:: recompute_fields(cr, model, fields[, ids=None][, logger=_logger][, chunk_size=256][, strategy="auto"])
+
+   Recompute field values
+
+   :param str model:  model name of the field(s) to recompute
+   :param list(str) fields: list of field names to recompute
+   :param list(int) ids: list of record IDs to recompute
+   :param logger: Logger used to print the progress of the function
+   :type: logger: :class:`logging.Logger`
```

Comment:
Sphinx doesn't seem to like alternating docstring styles. `Type` is rendered after all params.